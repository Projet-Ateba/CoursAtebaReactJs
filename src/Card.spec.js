import {expect} from 'chai'
import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";
import Sinon from 'sinon';


describe('<card />', () => {
    it('Vérifier que les `onClick` fonctionne dans nos prop quand on clique', () =>{
        const onClick = Sinon.spy()
        const wrapper = shallow(<Card card="🌍" feedback="hidden" index={0} onClick={onClick} />)
        wrapper.simulate('click')
        expect(onClick).to.have.been.calledWith(0)
    })

    it('doit correspondre au referent snapshot', ()=> {
        const onClick = Sinon.spy()
        const wrapper = shallow(<Card card="🌍" feedback="hidden" index={0} onClick={onClick} />)
        expect(wrapper).to.matchSnapshot()
    })
})