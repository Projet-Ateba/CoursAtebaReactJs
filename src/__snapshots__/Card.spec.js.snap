// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<card /> doit correspondre au referent snapshot 1`] = `
<div
  className="card hidden"
  onClick={[Function]}
>
  <span
    className="symbol"
  >
    ‚ùì
  </span>
</div>
`;
